### PR TITLE
DEVX-2185: "Data Masking" recipe migration

### PR DESCRIPTION
**Pull request by awalther28**
[Jira ticket](https://confluentinc.atlassian.net/browse/DEVX-2185)

This PR proposes migrating an original streams processing ["recipe" for data masking](https://www.confluent.io/stream-processing-cookbook/ksql-recipes/data-masking) to Kafka Tutorials. The semaphore test added for this KT is passing, no other KTs were altered in this PR. 

This is my first KT submission, please bear with me and thank you in advance 😄

### CODE REVIEW

**Review by bbejeck — COMMENTED** [review]
Thanks for PR @awalther28.
I have a few minor comments, otherwise this LGTM.  

Great job!

File: index.html [hunk 1]
```diff
@@ -70,6 +70,7 @@ <h2 class="subtitle">Apache Kafka is a powerful, scalable, fault-tolerant distri
             <li class="more is-hidden"><a href="dynamic-output-topic/kstreams.html">Choose the output topic dynamically</a></li>
             <li class="more is-hidden"><a href="naming-stateful-operations/kstreams.html">Name stateful operations</a></li>
             <li class="more is-hidden"><a href="kafka-streams-convert-to-ktable/kstreams.html">Convert a KStream to a KTable</a></li>
+            <li class="more is-hidden"><a href="masking-data/ksql.html">Masking data</a></li>
```
[the remark this answers is not in the document]
Comment (awalther28) [reply index.html]: Makes sense, I moved it to the top of the list.

File: _includes/tutorials/masking-data/ksql/markup/dev/print-purchases-obfuscated-topic.adoc [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+====
+There are a few types of masking functions and optional parameters that may be of use to you. +
+
+*Optional arguments:* +
+`MASK(CUSTOMER_NAME, 'X', 'x', 'n', '-')` +
+In the example above, the following types of characters in `CUSTOMER_NAME` would be masked: upper-case letters would become `X`, lower-case letters would become `x`, numbers would become `n`, and other characters would become `-`. This is the default setting if no mask characters are present. Set a given mask character to NULL to prevent any masking of that character type. +
+
+*Other types of `MASK` Functions:* +
+KSQL offers a variety of different masking functions that allow you to mask the farthest or nearest _x_ number of characters on right or left. Check out the KSQL documentation link:https://docs.ksqldb.io/en/latest/developer-guide/ksqldb-reference/scalar-functions/#mask[*link here*,window=_blank] for more information.
+====
```
[the remark this answers is not in the document]
Comment (awalther28) [reply _includes/tutorials/masking-data/ksql/markup/dev/print-purchases-obfuscated-topic.adoc:24]: A few other tutorials have something along the lines of ```Type 'exit' and hit enter to exit the ksqlDB cli.``` I just added a very similar statement to the end of the dev instructions. I can change this to ```CTRL+D``` if you prefer.

[the remark this answers is not in the document]
Comment (bbejeck) [reply _includes/tutorials/masking-data/ksql/markup/dev/print-purchases-obfuscated-topic.adoc:24]: >I can change this to CTRL+D if you prefer.

I like what you've done better.

**Review by bbejeck — APPROVED** [review]
Thanks for the quick response @awalther28, LGTM

**Pull request merged**